packages/manager/core/application/src/ovh-tracking.tsx

```diff
@@ -0,0 +1,131 @@
+import { useEffect, useLayoutEffect, useState, useRef } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useShell } from '.';
+
+// Send hit with data-tracking attributes
+const ODS_COMPONENTS = ['OSDS-BUTTON', 'OSDS-LINK'];
+// Send hit when value attributes change
+const ODS_COMPONENTS_GROUP = [
+  'osds-select',
+  'osds-radio-group',
+  'osds-checkbox',
+];
+
+export default function OvhTracking() {
+  const location = useLocation();
+  const shell = useShell();
+  const { tracking, environment } = shell;
+  const env = environment.getEnvironment();
+  const [locationPath, setLocationPath] = useState(location);
+  const myStateRef = useRef(locationPath);
+
+  // Use trackingPage to send action load
+  const OvhTrackPage = (loc: any) => {
+    const path = loc?.pathname.split('/')[1];
+    const partsReplace = loc.pathname.substring(1).replaceAll('/', '::');
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${universe}::${applicationName}::${partsReplace ||
+        'homepage'}`;
+      tracking.trackPage({
+        name,
+        level2: '81',
+        page_category: path || 'homepage',
+      });
+    });
+  };
+
+  useEffect(() => {
+    tracking.init(false);
+  }, []);
+
+  useEffect(() => {
+    // When location change, send tracking load page
+    OvhTrackPage(location);
+    myStateRef.current = location;
+    setLocationPath(location);
+  }, [location]);
+
+  const ovhTrackingSendCLick = (value: string) => {
+    const path = myStateRef.current.pathname.split('/')[1];
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${applicationName}::${path || 'homepage'}::${value}`;
+      const page = `${universe}::${applicationName}::${path || 'homepage'}`;
+      tracking.trackClick({
+        name,
+        page: { name: page },
+        complete_page_name: page,
+        level2: '81',
```

Comment:
**nitpick:** `level2` is set based on the universe. So it should not be hard-coded

Reply:
For the moment, we didn't have the key in the useShell(), that why I hard coded this

In the this file 'packages/components/ovh-at-internet/src/constants.ts' you have a constant AT_INTERNET_LEVEL2, but we can't get the key nowhere

Reply:
If the key is not present in `useShell()`, you can add it.